WordPress/Classes/ViewRelated/Post/PostCardActionBar.m

```diff
@@ -0,0 +1,367 @@
+        [button removeFromSuperview];
+    }
+    self.currentBatch = 0;
+    self.shouldShowMore = [self checkIfShouldShowMoreButton];
+    NSInteger numOfButtons = [self maxButtonsToDisplay];
+    numOfButtons = MIN(numOfButtons, [self.items count]);
+
+    NSMutableArray *buttons = [NSMutableArray array];
+    for (NSInteger i = 0; i < numOfButtons; i++) {
+        UIButton *button = [self newButton];
+        [self.contentView addSubview:button];
+        [buttons addObject:button];
+    }
+    self.buttons = buttons;
+
+    [self setupConstraints];
+}
+
+- (UIButton *)newButton
+{
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+    button.translatesAutoresizingMaskIntoConstraints = NO;
+    button.backgroundColor = [WPStyleGuide lightGrey];
+    button.titleLabel.font = [WPStyleGuide subtitleFont];
+    [button setTitleColor:[WPStyleGuide wordPressBlue] forState:UIControlStateNormal];
+    [button setTitleColor:[WPStyleGuide darkBlue] forState:UIControlStateHighlighted];
+    [button addTarget:self action:@selector(handleButtonTap:) forControlEvents:UIControlEventTouchUpInside];
+
+    return button;
+}
+
+- (void)setupButtonsIfNeeded
+{
+    // check if we need to show the more button.
+    if ([self checkIfShouldShowMoreButton] != self.shouldShowMore) {
+        // configure buttons
+        [self setupButtons];
+        [self configureButtons];
+    }
+}
+
+
+#pragma mark - Configuration
+
+- (void)configureButtons
+{
+    // Reset all buttons.
+    for (UIButton *button in self.buttons) {
+        [self configureButton:button forItem:nil atIndex:0];
+    }
+
+    NSArray *itemsToShow = [self currentBatchOfItems];
+    for (NSInteger i = 0; i < [itemsToShow count]; i++) {
+        PostCardActionBarItem *item = [itemsToShow objectAtIndex:i];
+        UIButton *button =[self.buttons objectAtIndex:i];
+        [self configureButton:button forItem:item atIndex:[self indexOfItem:item]];
+    }
+
+    if (self.shouldShowMore) {
+        BOOL isLastBatch = [itemsToShow lastObject] == [self.items lastObject];
+        PostCardActionBarItem *moreItem = [self moreItem:isLastBatch];
+        [self configureButton:[self.buttons lastObject] forItem:moreItem atIndex:ActionBarMoreButtonIndex];
+    }
+}
+
+- (void)configureButton:(UIButton *)button forItem:(PostCardActionBarItem *)item atIndex:(NSUInteger)index
+{
+    button.tag = index;
+    [button setTitle:item.title forState:UIControlStateNormal];
+    [button setTitle:item.title forState:UIControlStateHighlighted];
+    [button setImage:item.image forState:UIControlStateNormal];
+    [button setImageEdgeInsets:item.imageInsets];
+    [button setImage:item.highlightedImage forState:UIControlStateHighlighted];
+}
+
+- (void)configureButtonsWithAnimation
+{
+    // Frames for transtion
+    CGRect frame = self.contentView.frame;
+    CGRect smallFrame = frame;
+    smallFrame.origin.x += smallFrame.size.width / 2.0;
+    smallFrame.origin.y += smallFrame.size.height / 2.0;
```

Comment:
Just in case: are we sure that both Width + Height will always be even numbers?. Otherwise this OP might not result in an even number, and the button might end up a little bit blurry onscreen.